rust_src/src/buffers.rs

```diff
@@ -434,4 +468,36 @@ pub extern "C" fn nsberror(spec: Lisp_Object) -> ! {
     }
 }
 
+/// These functions are for debugging overlays.
+
+/// Return a pair of lists giving all the overlays of the current buffer.
+/// The car has all the overlays before the overlay center;
+/// the cdr has all the overlays after the overlay center.
+/// Recentering overlays moves overlays between these lists.
+/// The lists you get are copies, so that changing them has no effect.
+/// However, the overlays you get are the real objects that the buffer uses.
+#[lisp_fn]
+pub fn overlay_lists() -> LispObject {
+    let list_overlays = |ol: Option<LispOverlayRef>| -> LispObject {
+        let mut ol_list = LispObject::constant_nil();
+
+        if let Some(o) = ol {
+            let mut it = o.iter();
+
```

Comment:
The upcoming transmute call should really be a method on `LispOverlayRef`. This is basically `as_lisp_obj` right?

Reply:
I don't understand which part you mean. `iter` returns a `LispOverlayIter`.

Reply:
My comment is intended for the while loop. Which is why I said "upcoming transmute call". If I had put the comment on the transmute line the while loop would not be legible.